scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,618 @@
 #!/bin/sh
+### v 1.11.15-1
+###
+###############################
+###
+### Nix installation script
+### Universal shell script for for OS Linux environments
+###
+###############################
+###
+### Upstream URL: https://github.com/NixOS/nix/blob/master/scripts/install-nix-from-closure.sh
+###
+### Script is fully POSIX compatible, with controlled exceptions.
+### (must be) Fully transnational. That means, if it not succeeded - it does rolls back.
+###
+### `/bin/sh -u` is not possible to do, because Docker environments has USER='', and shell checks that and not get even to execution part
+###
+#
+###############################
+###  Documentation
+###############################
+{
+#
+#      Installer consist of:
+#      1. Setup environment
+#      2. Main constants
+#      3. CLI control constants
+#      4. CLI output functions
+#      5. Program stage functions
+#      6. Main function
+#
+#
+
+# Special things about this script
+#
+# 1) Notice, Warning, Error, ErrorRevert massages have special form to be very readable and get attention.
+# Message body starts from new line.
+# And has 4 spaces from the left. Always.
+#
+###############################
+#
+#    ^ notice "
+#    |
+#<---Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users independently.
+#    "
+#    fi
+#
+###############################
+#
+# This produces output in color:
+#
+###############################
+#
+#./install: Notice:
+#
+#    Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users independently.
+#
+###############################
+#
+true # so {...} is not empty, shell will not permit otherwise
+}
```

Comment:
It is for convenience to be able to shrink blocks of script with that braces in IDEs or editors.

Since script is relatively large, you can fold parts so you can observe main stages, and also to concentrate on what is important now.

Reply:
Aha ok. Like I said, no biggie.